core/src/avm1/object/stage_object.rs

```diff
@@ -681,15 +685,28 @@ fn set_x_scale<'gc>(
     val: Value<'gc>,
 ) -> Result<(), Error<'gc>> {
     if let Some(val) = property_coerce_to_number(activation, val)? {
+        let val = if activation.swf_version() < 5 {
+            let v = if this.scale_x(activation.context.gc_context).into_fraction() < 0.0 {
+                -val
+            } else {
+                val
+            };
+            (Fixed16::from_f64(v / 100.0).to_f64()) * 100.0
+        } else {
+            val
+        };
```

Comment:
The behavior shouldn't change based on positive or negative -- instead, SWFv4 does not cache the value after its set `DisplayObjectFlags::SCALE_ROTATION_CACHED`. It will naturally return positive once its calculated again.

The clamping to `Fixed16` part should happen all the time, even in SWFv5+. So that part should probably move to `DisplayObject::set_scale_x`.

For example:
```
clip._xscale = 0.00152587890625; // 100/65536
trace(clip.transform.matrix);
clip._xscale = 0.00152587890624; // <100/65536, clamps to 0
trace(clip.transform.matrix);
```